RFCs/0009-Fn::Map.md

```diff
@@ -0,0 +1,898 @@
+# New Intrinsic Function for looping: `Fn::Map`
```

Comment:
Should we include Fn::Merge in the title?

Reply:
I don't think it should be a separate RFC. The intention is to add looping functionality, so the essential components of that should be included here. I changed the title to `Looping functionality for CFN Templates`